Un'eccezione in programmazione è un evento anomalo o imprevisto che si verifica durante l'esecuzione di un programma e interrompe il flusso normale di esecuzione. Le eccezioni possono essere causate da errori di programmazione, errori di runtime o situazioni impreviste, come la mancanza di memoria o l'impossibilità di accedere a un file. L'OOP gestisce le eccezioni attraverso meccanismi come le istruzioni `try`, `catch` e `finally`, che consentono di gestire e recuperare da eccezioni durante l'esecuzione del programma. Con un blocco `try`, il codice che potrebbe generare un'eccezione viene racchiuso tra parentesi graffe e seguito da uno o più blocchi `catch` che gestiscono specificamente le eccezioni che possono verificarsi. Un blocco `finally` può essere utilizzato per eseguire del codice che deve essere eseguito indipendentemente dall'occorrenza di un'eccezione. La gestione delle eccezioni è importante perché consente di scrivere codice robusto che può gestire e recuperare da situazioni impreviste senza interrompere l'esecuzione del programma.
